tests/start.sh

```diff
@@ -8,16 +8,21 @@ MACHINE=${JOB_NAME#yocto-}
 DEVICE_TYPE_JSON=$WORKSPACE/$MACHINE.json
 DEPLOY_ARTIFACT=$(jq --raw-output '.yocto.deployArtifact' $DEVICE_TYPE_JSON)
 cp $(readlink --canonicalize $WORKSPACE/build/tmp/deploy/images/$MACHINE/$DEPLOY_ARTIFACT) $WORKSPACE/tests/autohat/resin.img
+AUTOHAT_IMAGE=autohat_$MACHINE\_$BUILD_NUMBER
+function cleanup {
+	docker rmi $AUTOHAT_IMAGE
+}
+docker build -t $AUTOHAT_IMAGE .
 
 if [ "$MACHINE" == "qemux86-64" ]; then
-	docker build -t autohat_$MACHINE\_$BUILD_NUMBER .
 	docker run --rm -v ${WORKSPACE}/tests/autohat:/autohat --privileged \
 		--env RESINRC_RESIN_URL=${RESINRC_RESIN_URL} \
 		--env email=${RESIN_EMAIL} \
 		--env password=${RESIN_PASSWORD} \
 		--env device_type=${MACHINE} \
 		--env application_name=${MACHINE//-} \
 		--env image=/autohat/resin.img \
-		autohat robot --exitonerror /autohat/example.robot
-	docker rmi autohat_$MACHINE\_$BUILD_NUMBER
+		$AUTOHAT_IMAGE robot --exitonerror /autohat/example.robot
 fi
+
+trap cleanup EXIT
```

Comment:
@agherzan please look at the over all change - I removed this with signals later on :) - Commit by commit is bad in this PR.